jungfrau_utils/jf_file.py

```diff
@@ -79,14 +79,17 @@ def __exit__(self, *args):
         self.close()
 
     def __getitem__(self, item):
-        if isinstance(item, int):
+        if isinstance(item, int) or isinstance(item, slice):
             ind, roi = item, ()
         else:
-            ind, roi = item[0], item[1:]
+            ind, *roi = item
+
+        if len(roi) > 2:
+            raise IndexError("too many indices for ROI")
 
         jf_data = self.jf_file['/data/{}/data'.format(self.detector_name)][ind]
 
-        if self.jf_handler.highgain != self.daq_rec[ind] & 0b1:
+        if (self.jf_handler.highgain != self.daq_rec[ind] & 0b1).any():
```

Comment:
'daq_rec', as well as 'module_map', are arrays and can be (theoretically) different for images in the same file, so this has to go into a sort of a loop over image indexes. However, it's not yet clear to me whether jf_file needs to take care of slicing, maybe JFDataHandler's api could be improved, so that jf_file can simply propagate slice params. 

@NichtJens , would be great to hear your ideas?

Reply:
Ahh .. OK. This needs to be done differently then.

You are certainly right: Slicing is complicated enough so that duplicating the functionality should not happen. If I have a good idea for a clean solution, I'll let you know ;)